### PR TITLE
Tooltip rui-components

### PR DESCRIPTION
**Pull request by jerseygooner**
Rui-Component for tooltips similar to the helptext functionality already available.

### CODE REVIEW

**Comment by ajkitson** [comment]
Should the rui-helptext be a wrapper for this new component?

File: src/js/directives/tooltip.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var app = angular.module('ruiComponents');
+
+app.directive('ruiTooltip', ['$compile', function ($compile) {
+	return {
+		restrict: 'A',
+		scope: {
+      message: '@',
+      data: '='
+    },
+    transclude: "element",
```
Comment (ajkitson) [inline src/js/directives/tooltip.js]: what's this do? I've only seen `transclude: true` before

File: src/js/directives/tooltip.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+var app = angular.module('ruiComponents');
+
+app.directive('ruiTooltip', ['$compile', function ($compile) {
```
Comment (ajkitson) [inline src/js/directives/tooltip.js]: why do we need `$compile`?

File: src/js/directives/tooltip.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    transclude: "element",
+    replace: true,
+    templateUrl: 'templates/tooltip.html',
+
+    link: function(scope, element, attrs, ctrl, linker){
+      if (scope.data){
+        scope.message = scope.data;
+      }
+    }
+	};
```
Comment (ajkitson) [inline src/js/directives/tooltip.js]: spacing issues in a handful of places in this file

File: src/templates/tooltip.html [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<span class="rui-tooltip-container" ng-mouseover="showtooltip=true" ng-mouseleave="showtooltip=false" ng-click="clicked=!clicked">
+  <span class="rui-tooltip" ng-class="{'rui-hidden': (!(clicked || showtooltip))}">{{message}}</span>
```
Comment (ajkitson) [inline src/templates/tooltip.html]: If I hover and then click, the tooltip stays around. Don't think that's what we want. Do we need the `ng-click` at all, given that, if you're clicking you're already hovering?

File: src/js/directives/tooltip.js (hunk 1 above)
Comment (jerseygooner) [inline src/js/directives/tooltip.js]: "With regular transclusion it is the descendants of the current element that will be transcluded. With element transclusion it is the element itself (including all of its descendants)."

In this case, it doesn't really matter - so I've switched this to 'true' as that is more commonly used and keeping the original element around makes sense here.

File: src/js/directives/tooltip.js (hunk 2 above)
Comment (jerseygooner) [inline src/js/directives/tooltip.js]: Deleted.

File: src/js/directives/tooltip.js (hunk 3 above)
Comment (jerseygooner) [inline src/js/directives/tooltip.js]: Beatified :)

**Comment by jerseygooner** [comment]
I've made helptext a wrapper for tooltip.

File: src/templates/tooltip.html (hunk 4 above)
Comment (jerseygooner) [inline src/templates/tooltip.html]: That is the functionality from helptext. Given these are so closely related, I think we should keep the functionality in sync - take it or leave it IMO.

**Comment by ajkitson** [comment]
Help text is now bold...
image

File: src/templates/tooltip.html (hunk 4 above)
Comment (ajkitson) [inline src/templates/tooltip.html]: Let's take it out then. When I run into it, it feels like a bug.

File: src/templates/helptext.html [hunk 5]
```diff
@@ -1,6 +1,4 @@
 <div class="rui-helptext-container">
-  <span class="rui-helptext-icon ion-help-circled" ng-mouseover="showtooltip=true" ng-mouseleave="showtooltip=false" ng-click="clicked=!clicked" >
-    <span class="rui-tooltip" ng-class="{'rui-hidden': (!(clicked || showtooltip))}">{{message}}</span>
-  </span>
+  <span class="rui-helptext-icon ion-help-circled" rui-tooltip message={{message}}></span>
```
Comment (ajkitson) [inline src/templates/helptext.html]: Nice and DRY :+1:

**Comment by ajkitson** [comment]
Back at you @jerseygooner Just two things:
- the helptext/tooltip seems to inherit the formatting of the element to which the tooltip was added, at least if there are `<b>` tags. Probably something we can do with how the transclusion works in the element
- let's get rid of the thing where clicking the element that has the tooltip causes it to hang around

File: src/templates/tooltip.html (hunk 4 above)
Comment (jerseygooner) [inline src/templates/tooltip.html]: Sounds good - I removed it.

**Comment by jerseygooner** [comment]
Address both items. 
1. Hard coded the tooltip text to be normal font-weight. 
2. Removed the clickable tooltip stuff.

**Comment by ajkitson** [comment]
looks good

**Pull request merged**